backends-common/cassandra/src/main/java/org/apache/james/backends/cassandra/utils/CassandraAsyncExecutor.java

```diff
@@ -66,7 +68,7 @@ public CassandraAsyncExecutor(Session session) {
 
     public Mono<Row> executeSingleRow(Statement statement) {
         return executeSingleRowOptional(statement)
-                .handle((t, sink) -> t.ifPresent(sink::next));
+                .handle(unboxOptional());
```

Comment:
It's not really unboxing ... it rather emit the value if there's some ... if don't know how to call that, but without a good naming, I prefer the inlined code

Reply:
unwrapOptional?

Technical code deduplication is a good thing. IMO we fail to hide technical details of reactor defaults, setting higher the reactive entry bar in our code.

Making writting the efficient easier will ensure people will use it more.

Reply:
A very important thing is that we use common knowledge as much as possible. I mean, when something generic has a given name in the computer science domain, we should use that name.
If it doesn't exist, we should ask ourselves why it doesn't exist and look for common solution to that problem (but not create ad hoc solution has much as possible).

To analyze your suggestion: unwrap  would mean we have an object A wrapper in an object B.
Here, A is generic and B is Optional.
So B not always wrap an A but it **may** contain a A (wrapping would mean it let the owner of the object to call features of that object and ) and it may not.
So I don't agree with `wrap` semantic and the method name is not helping understand how is empty handled.
maybe `publishIfPresent`